saleor/checkout/utils.py

```diff
@@ -871,6 +876,19 @@ def create_order(checkout: Checkout, tracking_code: str, discounts, taxes):
 
     # assign checkout payments to the order
     checkout.payments.update(order=order)
+
+    # remove checkout after order is created
+    checkout.delete()
+
+    # Create the order placed and email confirmation sent events
+    OrderEventManager()\
```

Comment:
I agree about the last part, this might be more helpful it the event was created from the same task so we are actually sure that the task was executed.